# HG changeset patch
# User Rik <rik@octave.org>
# Date 1489199570 28800
#      Fri Mar 10 18:32:50 2017 -0800
# Node ID c3c26f6a78121bff845f959cbe1e751eaf77e792
# Parent  be761cce7de09fcaf876eb8fade1740c6a2b16d5
__gnuplot_draw_axes__.m: Use try/catch rather than unwind protect in 589e4e4548e1.

* __gnuplot_draw_axes__.m: Use try/catch rather than unwind protect around
temporary fix for legends in gnuplot.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -503,22 +503,21 @@ function __gnuplot_draw_axes__ (h, plot_
     h_obj = kids(end);
     kids = kids(1:(end-1));
 
     obj = get (h_obj);
 
     ## FIXME: Temporary workaround to add hidden interpreter property back.
     ##        Delete when bug #50496 is fixed.
     wstate = warning ("query", "Octave:deprecated-property");
-    unwind_protect 
+    try 
       warning ("off", "Octave:deprecated-property");
       obj.interpreter = get (h_obj, "interpreter");
-    unwind_protect_cleanup
-      warning (wstate);
-    end_unwind_protect
+    end_try_catch
+    warning (wstate);
 
     if (isfield (obj, "xdata"))
       obj.xdata = double (obj.xdata);
     endif
     if (isfield (obj, "ydata"))
       obj.ydata = double (obj.ydata);
     endif
     if (isfield (obj, "zdata"))
